subsys/bluetooth/host/gatt.c

```diff
@@ -5896,13 +5926,30 @@ static int cf_set(const char *name, size_t len_rd, settings_read_cb read_cb,
 	}
 
 	if (len_rd) {
-		len = read_cb(cb_arg, cfg->data, sizeof(cfg->data));
+		len = read_cb(cb_arg, dst, sizeof(dst));
 		if (len < 0) {
 			LOG_ERR("Failed to decode value (err %zd)", len);
 			return len;
 		}
 
+		memcpy(cfg->data, dst, sizeof(cfg->data));
 		LOG_DBG("Read CF: len %zd", len);
+
+		if (len != sizeof(dst)) {
+			LOG_DBG("Change-aware status not stored, "
+				"defaulting peer status to change-aware");
+			set_change_aware(NULL, cfg, true);
```

Comment:
Good point, I reverted it to change-unaware.

For the bit, I prefer if that bit is set from only one place. 
I removed the manual setting below, since in that case (when running `settings_load`) we will not be connected, and `set_change_aware` will only set the bit and not try to write it back to NVS.

`set_change_aware` needs to be able to have `conn == NULL` because it can get called from other places where the connection is not valid anymore.